### ISSUE TITLE
ERROR: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:28284

### ISSUE THREAD

**Issue by JonnyLinja** [Microsoft/vscode-react-native#485]
## My Steps
1. I have Reactotron installed and integrated with Redux within the code (not sure if that makes any difference, but figured it worth noting)
2. I have Reactotron open
3. I attempt to debug the app, the app opens
4. I then select `Debug JS Remotely` from the `Developer Menu`

## Actual Behavior
1. It closes the app and displays the error `Ensure Node was launched with --inspect. Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:39906).`
2. Trying to open the app again even in just the simulator will white screen, hang, then crash.
3. If I attempt to open the `Developer Menu` during the white screen, it will immediately crash

## Expected Behavior
1. I can debug the app

## Software versions
- React-Native VS Code extension version: `0.3.2`
- VSCode version: `1.14.2`
- OS platform and version: `Mac OS 10.12.6`
- NodeJS version: `v8.2.1`
- React Native version: `0.46.4`

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
[vscode-react-native] Starting React Native Packager.
[vscode-react-native] Prewarming bundle cache. This may take a while ...
[vscode-react-native] Building and running application.
[vscode-react-native] Executing command: react-native run-ios --simulator iPhone 5s --project-path ios
Scanning 529 folders for symlinks in /Users/Administrator/Documents/OpenBarbell/OpenBarbellApp/node_modules (7ms)
Found Xcode project OBBApp.xcodeproj
Launching iPhone 5s (iOS 10.3)...
Building using "xcodebuild -project OBBApp.xcodeproj -configuration Debug -scheme OBBApp -destination id=7DED4844-C814-41F8-98C5-6E5979E3CB89 -derivedDataPath build"

...

** BUILD SUCCEEDED **


Installing build/Build/Products/Debug-iphonesimulator/OBBApp.app
Launching com.openbarbellapp
com.openbarbellapp: 10521
[vscode-react-native] Finished executing: react-native run-ios --simulator iPhone 5s --project-path ios
[vscode-react-native] Executing command: react-native run-ios --simulator iPhone 5s --project-path ios
Scanning 529 folders for symlinks in /Users/Administrator/Documents/OpenBarbell/OpenBarbellApp/node_modules (7ms)
Found Xcode project OBBApp.xcodeproj
We couldn't boot your defined simulator due to an already booted simulator. We are limited to one simulator launched at a time.
Launching iPhone 5s (iOS 10.3)...
Building using "xcodebuild -project OBBApp.xcodeproj -configuration Debug -scheme OBBApp -destination id=7DED4844-C814-41F8-98C5-6E5979E3CB89 -derivedDataPath build"

...

** BUILD SUCCEEDED **


Installing build/Build/Products/Debug-iphonesimulator/OBBApp.app
Launching com.openbarbellapp
com.openbarbellapp: 10597
[vscode-react-native] Finished executing: react-native run-ios --simulator iPhone 5s --project-path ios
[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
[vscode-react-native] Debugger worker loaded runtime on port 39906
```
- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
Getting exponent dependecy.
######### Checking react native is correctly setup. This may take a while... ##########
Checking if the correct react native is installed..
######### Starting Packager ##########
######### Executing command: react-native start --port 8081 ##########
Scanning 529 folders for symlinks in /Users/Administrator/Documents/OpenBarbell/OpenBarbellApp/node_modules (7ms)
 ┌────────────────────────────────────────────────────────────────────────────┐ 
 │  Running packager on port 8081.                                            │ 
 │                                                                            │ 
 │  Keep this packager running while developing on any JS projects. Feel      │ 
 │  free to close this tab and run your own packager instance if you          │ 
 │  prefer.                                                                   │ 
 │                                                                            │ 
 │  https://github.com/facebook/react-native                                  │ 
 │                                                                            │ 
 └────────────────────────────────────────────────────────────────────────────┘ 
Looking for JS files in
   /Users/Administrator/Documents/OpenBarbell/OpenBarbellApp 


React packager ready.

Loading dependency graph, done.
######### Packager started. ##########
Bundling `index.ios.js`  100.0% (920/920), done.
######### The Bundle Cache was prewarmed. ##########
Bundling `index.ios.js`  100.0% (920/920), done.
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/Administrator/Documents/OpenBarbell/OpenBarbellApp"
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29 [Extension Host] Error: The module '/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 57. This version of Node.js requires
NODE_MODULE_VERSION 53. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or`npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Function.G.o._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:399199)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
h.logExtensionHostMessage @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29 [Extension Host] Error: Cannot find module './build/default/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.G.o._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:399199)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
h.logExtensionHostMessage @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29 [Extension Host] Error: Cannot find module './build/Debug/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.G.o._load (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:23:399199)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/Administrator/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
h.logExtensionHostMessage @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29 [Extension Host] (node:10322) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
h.logExtensionHostMessage @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:29
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28 Ensure Node was launched with --inspect. Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:39906).
e.doShow @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:28
```

**Comment by vladimir-kotikov** [comment]
@JonnyLinja, thanks for reporting. We just published a new version of extension (0.4.0) - could you please try to update and see if the issue still reproduces?

**Comment by vladimir-kotikov** [comment]
@JonnyLinja, any updates? Has this been fixed for you after upgrading extension? Though I remember I've seen this before but this was happening quite sporadically I wasn't able to repro this on 0.3.2, 0.4.0 and latest dev versions.

**Comment by JonnyLinja** [comment]
@vladimir-kotikov I still have this issue on 0.4.0

**Comment by vladimir-kotikov** [comment]
@JonnyLinja I wasn't able to repro this - I have exactly the same versions of everything but still can both start debugging and attach to already running packager without any issues. I might be missing something, so could you please post _exact and detailed_ repro steps to make sure I did everything right :)

Thanks

**Comment by JonnyLinja** [comment]
Hmm.

So I tried again with my actual project and I still had the err could not connect issues.

Then I tried breakpoints on a blank project that I had created previously. I no longer got the connection issues. Debugging would start, but it wouldn't stop on any of the breakpoints despite breakpoints being listed as verified.

I then recreated the blank project, and everything was working on that.

I'm attempting to upgrade my actual project to React 0.47.1 to see if that makes any difference. Currently it isn't building and I'm having issues with npm install that I'm trying to work through.

So for the debugging itself, I suspect it's...something with my actual project, be it my code, xcodeproj problems, or one of my dependencies. I'm just not sure how to actually go about debugging it.
